hermes/crates/cardano-chain-follower/examples/load_snapshot.rs

```diff
@@ -0,0 +1,60 @@
+//! Test
+use std::{
+    error::Error,
+    path::PathBuf,
+    time::{Duration, Instant},
+};
+
+use pallas::ledger::traverse::MultiEraBlock;
+use pallas_hardano::storage::immutable::Point;
+use tracing::level_filters::LevelFilter;
+use tracing_subscriber::EnvFilter;
+
+/// Main test
+#[allow(clippy::unwrap_used)]
+fn main() -> Result<(), Box<dyn Error>> {
+    tracing_subscriber::fmt()
+        .with_env_filter(
+            EnvFilter::builder()
+                .with_default_directive(LevelFilter::INFO.into())
+                .from_env_lossy(),
+        )
+        .init();
+    let path = PathBuf::from(std::env::var("CARGO_MANIFEST_DIR")?).join("examples/base_snapshot");
+    let tip = pallas_hardano::storage::immutable::get_tip(&path)?.ok_or("bad");
+    println!("{path:?} {tip:?}");
+
+    let start_point = Point::Origin;
+    let start_time = Instant::now();
+    let total_slots = tip.unwrap().slot_or_default();
+    let mut block_data_iter =
+        pallas_hardano::storage::immutable::read_blocks_from_point(&path, start_point)?;
+
+    let timeout_duration = Duration::from_secs(10);
+
+    let mut timer = Instant::now();
+    let mut slots_seen = 0;
+    while let Some(Ok(block_data)) = block_data_iter.next() {
+        let block = MultiEraBlock::decode(&block_data)?;
+        slots_seen += 1;
+        if timer.elapsed() >= timeout_duration {
+            let total_duration = start_time.elapsed().as_secs();
+            tracing::info!(
+                slots_seen = slots_seen,
+                total_time = total_duration,
+                "NUMBER={} SLOT={} HASH={}",
+                block.number(),
+                block.slot(),
+                hex::encode(block.hash())
+            );
+            timer = Instant::now();
+        }
```

Comment:
The stats seem off.  `slots_seen=2118401 total_slots_expected=56722224` looks like we missed data/
is it that `slots_seen` should be `blocks_seen`?
Can we also make the output columnar, and add summary data to the line, such as slots/second, blocks/second (aggregate since the start, and also just for the current interval.)

For example more like this:
```
BLOCK   |   SLOT    |   Total Time |  Blocks/sec | blocks/sec/int | slots/sec | slots/sec/int |
792223 |  24833596 | 00:00:10 | 79222.3 | 79222.3 | 2483359.6 | 2483359.6 | 
1318069 | 37574044 | 00:00:20 | 65,903.45 | 52584.6 | 1878702.2 | 1274044.8 | 
```
And so on, but pad the columns so they are better aligned.
`blocks/sec/int` means blocks/second in the last interval or since the last time we printed.  The other stat is cumulative since the start of the run.

That data in the example is from my run of this test.  I want the summarily stats so I don;t have to guess how its performing over time or manually calculate them.

It does looks like its actually getting slower over time.  

To determine if that's the case, also add up the total size of all the blocks, and include a 
`block size` metric, and a `block MB/sec` and `block MB/sec/int` summary metric to the line.
It could be early blocks are tiny, so we can do lots but later they are big, so that's more data to process.
If the number of blocks/slots changes but the MB/s metrics don't we know that's the cause.

The data set is also not big enough to determine that.
Lets add a "prod" snapshot target to cat-ci which we can also use to run this test against.
The URL you can use for it is: https://storage.googleapis.com/cdn.aggregator.release-mainnet.api.mithril.network/mainnet-e477-i5604.d11e9b0a3cda3f32233cf7c89f0a1aa321182c790b3277af468c8051007f89b7.tar.zst
Note: that's 48GB so for now we would only use that in local testing, not in CI.  But we can still have the target in earthly.

You can different targets in the earthfile which use a different snapshot.

As you have made this a stand alone app, which I am not opposed to, I think we can just extend this to have a simple option (no need for complex cli, just read the first parameter) to make it run `pallas` or `hermes` mode.

`pallas` does what you do here.
`hermes` mode uses the chain follower crate to do exactly the same thing.

The metrics can be the same then, the only difference is how you source the blocks.

Reply:
Sorry, I hadn't uploaded the most recent code, I'll take a look at your comments and address them.